### DISCUSSION TITLE
Couple Questions

### DISCUSSION THREAD

**Discussion by maravelous** [probonopd/MiniDexed#729] [Q&A] [answered]
Hello, I'm new to this very neat little synth. Couple quick questions though:

Is the zero 2 robust enough to run it with all the voices? The wording on this is kinda vague cause first the zero and zero 2 are talked about and then it just says they won't all run on the zero, which could be referring to the zero 1 or to the zero line(1&2). Be pretty neat if it can :-). I definitely plan to use a pcm5122 i2s dac so no worries about hdmi passthrough

Also, can more screens and buttons be added? I know it's been suggested that one could just move on to microdexed, but I'm stubborn, lol. What I'm imaging is a screen and encoder for each of the dx7 instances, so that a performer can see what each is set to and change them independently. I'm not certain cause I haven't gotten to play with a microdexed, but it seems like if you wanted to do what I'm taking about there'd be a lot of quicky twisty between intention and execution.

 If the prescribed hardware can't handle that, I'm considering building an arduino midi controller and combining them both into the same box

Oh, and if the zero doesn't work, will the new lower ram 1gb pi 4 be sufficient or is that cutting the ram too aggressively? 

Thank you for any help you can provide. Bye!

**Comment by Banana71** [comment, marked as answer]
The Raspberry Pi Zero 2 can play all 8 TGs including reverb and compressor. miniDexed doesn't need much RAM and the RPi 4 1GB is well suited.

**Reply by probonopd** [reply]
Yes, I'd say it's currently the price-size-performance sweetspot.

**Reply by Banana71** [reply]
> Yes, I'd say it's currently the price-size-performance sweetspot.

Closely followed by the Raspberry Pi 3 Model A+. It has a standard HDMI and USB port

**Comment by probonopd** [comment]
> Also, can more screens and buttons be added?

Only one screen can be added, and even if you could add more than one, they would all show the same.
Having one screen per tone generator is not planned, even though I understand it might be neat for certain situations. But you can do that using a secondary MCU that sends MIDI commands to MiniDexed, if you configure each TG to be on its own MIDI channel.

**Reply by maravelous** [reply]
Thank you that was the work around I was thinking of if there wasn’t the option of adding the other screens directly to the mini dexed. 

So, and I think I know the answer but, could one forgo putting the screen and the encoder on the mini dexed and still be able to control all the same things with a custom built MIDI controller essentially serving as the user interface? I’d probably have the screen and encoder on their during prototyping, but if this is possible I might only use the feedback screens I set up for the midi controller and then package it all in one box

In my mind this would make my implementation 3 parts
1. Sound engine: Pi ,probably a zero 2 w/ audio hat, running mini dexed
2. Control interface: MIDI controller probably on arduino but maybe a teensy with buttons knob’s and sliders set to send the midi messages needed to control the sound engine
3. User feedback: larger screen set to display the elements selected by the control interface

This would present a possible failure point if I don’t adequately explore and map the mini dexed system or if something is missed in implementing that map to the control interface

But it would also allow, with not a ton more programming, the addition of a sequencer 

Could be missing something but I think that could be achieved with only another $50 ish dollars (not including the case I’ll 3d print)

I know this sounds a lot like a micro dexed now, but it should actually be quite a lot more powerful with the sound engine running on bare metal pi and not using any of the sound engines computing power to manage the user interface

**Reply by probonopd** [reply]
Might also want to look into https://codeberg.org/positionhigh/MicroDexed-touch. Doesn't use a Raspberry Pi though.

**Reply by diyelectromusic** [reply]
We don't have full MIDI control of all the functions at the moment.  There are quite a few though, so I guess it depends what you are planning on doing.  Have a look at the MIDI implementation page on the wiki (and ultimately, the defines at the start of the mididevice.cpp source file!).

We're still chewing over how to maybe open up the rest of MiniDexed to MIDI.  There are several options (see the discussion here for example: https://github.com/probonopd/MiniDexed/pull/704#issuecomment-2380427809)

Kevin

**Comment by Kirtai** [comment]
Note that the Zero 2 is roughly as powerful as a Pi3.
